include/range/v3/range/access.hpp

```diff
@@ -197,9 +205,8 @@ namespace ranges
         CPP_concept_bool has_member_end =
             CPP_requires ((T &) t) //
             (
-                _end_::is_sentinel<_begin_::_t<CPP_type(T) &>>(t.end())
-            ) &&
-            std::is_lvalue_reference<T>::value;
+                _end_::is_sentinel<_begin_::_t<CPP_type(T)>>(CPP_fwd(t).end())
```

Comment:
Both `CPP_fwd(t)` and `t` are lvalues of type `t` - this doesn't seem to change anything.

Reply:
Fixed this (and the next one) to actually be forwarding refs.